src/main/resources/python/security_schemes.hbs

```diff
@@ -172,37 +231,175 @@ class MutualTLSSecurityScheme(__SecuritySchemeBase):
+
+    @property
+    def auth_or_token_url(self) -> parse.ParseResult:
+        return self.token_url
+
+    def apply_auth(
+        self,
+        headers: _collections.HTTPHeaderDict,
+        resource_path: str,
+        method: str,
+        body: typing.Optional[typing.Union[str, bytes]],
+        query_params_suffix: typing.Optional[str],
+        client_info: OauthClientInfo,
+        scope_names: typing.Tuple[str, ...] = ()
+    ) -> None:
+        token = self._scope_names_to_token.get(scope_names)
+        if token:
+            super().set_token(headers, token)
+            return
+        client = self._scope_names_to_client.get(scope_names)
+        if client is None:
+            client = requests_client.OAuth2Session(
+                client_info.client_id,
+                client_info.client_secret,
+                scope=scope_names
+            )
+            self._scope_names_to_token[scope_names] = client
+        token = client.fetch_token(
+            url=parse.urlunparse(self.token_url),
+            username=self.username,
+            password=self.password
+        )
+        self._scope_names_to_token[scope_names] = token
+        print(token)
+        super().set_token(headers, token)
 
 
 @dataclasses.dataclass
-class TokenUrlOauthFlow:
-    token_url: str
+class ClientCredentialsOauthFlow(OAuthFlowBase):
+    token_url: parse.ParseResult
     scopes: typing.Dict[str, str]
     refresh_url: typing.Optional[str] = None
+    _scope_names_to_client: typing.Dict[typing.Tuple[str, ...], requests_client.OAuth2Session] = dataclasses.field(default_factory=dict)
+    _scope_names_to_token: typing.Dict[typing.Tuple[str, ...], OauthToken] = dataclasses.field(default_factory=dict)
+
+    @property
+    def auth_or_token_url(self) -> parse.ParseResult:
+        return self.token_url
+
+    def apply_auth(
+        self,
+        headers: _collections.HTTPHeaderDict,
+        resource_path: str,
+        method: str,
+        body: typing.Optional[typing.Union[str, bytes]],
+        query_params_suffix: typing.Optional[str],
+        client_info: OauthClientInfo,
+        scope_names: typing.Tuple[str, ...] = ()
+    ) -> None:
+        token = self._scope_names_to_token.get(scope_names)
+        if token:
+            super().set_token(headers, token)
+            return
+        client = self._scope_names_to_client.get(scope_names)
+        if client is None:
+            client = requests_client.OAuth2Session(
+                client_info.client_id,
+                client_info.client_secret,
+                scope=scope_names
+            )
+            self._scope_names_to_token[scope_names] = client
+        token = client.fetch_token(
+            url=parse.urlunparse(self.token_url),
+            grant_type='client_credentials'
+        )
+        self._scope_names_to_token[scope_names] = token
+        print(token)
```

Comment:
Please remove this print statment

Reply:
I agree. It should be developed in the future.